src/js_native_api.h

```diff
@@ -102,6 +102,34 @@ node_api_symbol_for(napi_env env,
                     size_t length,
                     napi_value* result);
 #endif  // NAPI_EXPERIMENTAL
+
+#ifdef NAPI_EMBEDDING
+NAPI_EXTERN napi_status NAPI_CDECL napi_create_platform(int argc,
```

Comment:
Currently the Node-API is split up between two files: `js_native_api.h` and `node_api.h`.
The difference is that the `js_native_api.h` contains only cross-engine JS-specific APIs and `node_api.h` has Node.JS-specific APIs. 
We use `js_native_api.h` in different scenarios where we want to interact with different JS engine using ABI-safe Node-API functions. For example, see the [Babylon Native](https://github.com/BabylonJS/BabylonNative/tree/master/Dependencies/napi), [Hermes Node-API](https://github.com/microsoft/hermes-windows/blob/main/API/hermes/hermes_napi.cpp) or [V8-JSI](https://github.com/microsoft/v8-jsi/tree/master/src/napi). In all these scenarios we implement `js_native_api.h` APIs, and we do not use Node.JS or Node-specific `node_api.h`. Even the `js_native_api_v8.h` and `js_native_api_v8.cc` must be free from any Node.JS specific code. Any interaction with the Node-JS specific code must be done either through the functions declared in the `js_native_api_v8_internals.h` or overriding virtual `napi_env__` in the derived `node_napi_env__` class in the `node_api_internals.h`.
The `node_api.h` has the Node.js specific functions and mostly contains code concerning Node-JS modules and asynchronous code.
From this perspective, the Node.JS embedding API must not be part of the `js_native_api.h` header file. It could be in the `node_api.h`, but since this file is to be used by modules, the best option is to put these new APIs in their own `node_hosting_api.h` or `node_embedding_api.h` file. Other applications, such as [Deno](https://deno.land/), could implement the full support for `node_api.h` and `js_native_api.h` to reuse the Node-API modules, but they would never implement the same Node.JS embedding APIs.
These embedding APIs are Node.JS specific and cannot be re-used for other engines or applications. E.g. other engines do not create a thread pool on their own, or cannot handle Node.JS CLI arguments,
It could be either pa